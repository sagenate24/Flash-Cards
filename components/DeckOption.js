import React from 'react';
import {
  View,
  Text,
  StyleSheet,
} from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

export default function DeckOption({
  size, name, iconStyle, subHeaderColor, children,
}) {
  return (
    <View>
      <Text style={styles.header}>
        <MaterialCommunityIcons
          size={size}
          name={name}
          style={iconStyle}
        />
      </Text>
      <Text style={[styles.subHeader, subHeaderColor]}>
        {children}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  header: {
    textAlign: 'center',
    paddingTop: 10,
    paddingBottom: 10,
  },
  subHeader: {
    fontSize: 14,
    textAlign: 'center',
    marginBottom: 5,
  },
});
