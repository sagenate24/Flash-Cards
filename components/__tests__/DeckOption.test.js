import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import DeckOption from '../DeckOption';

test('DeckOption snapShot', () => {
  const snap = renderer.create(
    <DeckOption size={34} name='cards-outline' iconStyle={{ color: '#000'}} subHeaderColor={{ color: '#fff' }}/>
  ).toJSON();

  expect(snap).toMatchSnapshot()
})