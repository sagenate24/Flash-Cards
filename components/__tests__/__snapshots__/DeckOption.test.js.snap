// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckOption snapShot 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "paddingBottom": 10,
        "paddingTop": 10,
        "textAlign": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "marginBottom": 5,
          "textAlign": "center",
        },
        Object {
          "color": "#fff",
        },
      ]
    }
  />
</View>
`;
